op-service/eth/transactions.go

```diff
@@ -129,14 +133,80 @@ type GenericTx interface {
 
```

Comment:
We should make sure to have a nice godoc that explains a `GenericTx` represents the consensus representation of a transaction from the perspective of the derivation pipeline with an explicit note of the EIP-2718 tx types that it supports. We will need to backport this to the specs, it can go in the base derivation pipeline doc

Reply:
Specs work happening here: https://github.com/ethereum-optimism/specs/pull/518